src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3531,69 +3537,63 @@ namespace winrt::Microsoft::Terminal::Control::implementation
-        // UI-driven, so that's okay. We're not really changing the scrollbar,
-        // but we do want to update the position of any search marks. The Core
-        // might send a scrollbar update event too, but if the first search hit
-        // is in the visible viewport, then the pips won't display until the
-        // user first scrolls.
-        auto scrollBar = ScrollBar();
-        ScrollBarUpdate update{
-            .newValue = scrollBar.Value(),
-            .newMaximum = scrollBar.Maximum(),
-            .newMinimum = scrollBar.Minimum(),
-            .newViewportSize = scrollBar.ViewportSize(),
-        };
-        _throttledUpdateScrollbar(update);
+        if (!_searchBox || _searchBox->Visibility() != Visibility::Visible)
+        {
+            return;
+        }
+
+        const auto text = _searchBox->Text();
+        if (text.empty())
+        {
+            return;
+        }
+
+        const auto goForward = _searchBox->GoForward();
+        const auto caseSensitive = _searchBox->CaseSensitive();
+        _handleSearchResults(_core.Search(text, goForward, caseSensitive, true));
     }
 
-    // Method Description:
-    // - Called when the core raises a UpdateSearchResults event. That's done in response to:
-    //   - starting a search query with ControlCore::Search.
-    //   - clearing search results due to change in buffer content.
-    // - The args will tell us if there were or were not any results for that
-    //   particular search. We'll use that to control what to announce to
-    //   Narrator. When we have more elaborate search information to report, we
-    //   may want to report that here. (see GH #3920)
-    // Arguments:
-    // - args: contains information about the search state and results that were
-    //         or were not found.
-    // Return Value:
-    // - <none>
-    winrt::fire_and_forget TermControl::_coreUpdateSearchResults(const IInspectable& /*sender*/, Control::UpdateSearchResultsEventArgs args)
+    void TermControl::_handleSearchResults(SearchResults results)
     {
-        co_await wil::resume_foreground(Dispatcher());
-        if (auto automationPeer{ Automation::Peers::FrameworkElementAutomationPeer::FromElement(*this) })
+        if (!_searchBox)
         {
-            automationPeer.RaiseNotificationEvent(
-                Automation::Peers::AutomationNotificationKind::ActionCompleted,
-                Automation::Peers::AutomationNotificationProcessing::ImportantMostRecent,
-                args.FoundMatch() ? RS_(L"SearchBox_MatchesAvailable") : RS_(L"SearchBox_NoMatches"), // what to announce if results were found
-                L"SearchBoxResultAnnouncement" /* unique name for this group of notifications */);
+            return;
         }
 
-        _UpdateSearchScrollMarks();
+        _searchBox->SetStatus(results.TotalMatches, results.CurrentMatch);
 
-        if (_searchBox)
+        if (results.SearchInvalidated)

```

Comment:
This avoids doing a `RaiseNotificationEvent` even though the search mask hasn't changed.